docs/core/diagnostics/metrics-collection.md

```diff
@@ -0,0 +1,404 @@
+---
+title: Collect metrics - .NET
+description: Tutorial to collect metrics in .NET applications
+ms.topic: tutorial
+ms.date: 10/27you/2021
+---
+
+# Collect metrics
+
+**This article applies to: ✔️** .NET Core 3.1 and later versions **✔️** .NET Framework 4.5 and later versions
+
+Instrumented code can record numeric measurements, but the measurements usually need to be aggregated, transmitted,
+and stored to create useful metrics for monitoring. This process of aggregating, transmitting and storing the data is
+called collection. In this tutorial, we will show several examples on how to collect metrics:
+  - Populating metrics in Grafana with OpenTelemetry and Prometheus
+  - Viewing metrics in real-time with the dotnet-counters command-line tool
+  - Creating a custom collection tool using the underlying .NET <xref:System.Diagnostics.Metrics.MeterListener> API
+
+For more information on custom metric instrumentation see [here](./understanding-different-metric-apis.md) for an overview
+of instrumentation options.
+
+### Create an example application
+
+**Prerequisites**: [.NET Core 3.1 SDK](https://dotnet.microsoft.com/download/dotnet) or a later version
+
+Before metrics can be collected, we need to produce some measurements. For simplicity we will create a small app that has
+some trivial metric instrumentation. The .NET runtime also has [a variety of metrics of built-in](available-counters.md). See 
+[the instrumentation tutorial](metrics-instrumentation.md) for more details about creating new metrics using the
+<xref:System.Diagnostics.Metrics.Meter?displayProperty=nameWithType> API shown here.
+
+```dotnetcli
+dotnet new console
+dotnet add package System.Diagnostics.DiagnosticSource
+```
+
+Replace the code of Program.cs with:
+```C#
+using System;
+using System.Diagnostics.Metrics;
+using System.Threading;
+
+class Program
+{
+    static Meter s_meter = new Meter("HatCo.HatStore", "1.0.0");
+    static Counter<int> s_hatsSold = s_meter.CreateCounter<int>("hats-sold"); 
+
+    static void Main(string[] args)
+    {
+        Console.WriteLine("Press any key to exit");
+        while(!Console.KeyAvailable)
+        {
+            // Pretend our store has a transaction each second that sells 4 hats
+            Thread.Sleep(1000);
+            s_hatsSold.Add(4);
+        }
+    }
+}
+```
+
+### Viewing metrics with dotnet-counters
+
+[dotnet-counters](dotnet-counters.md) is a simple command-line tool that can view live metrics for any .NET Core application on
+demand. It doesn't require any advance setup which can make it very useful for ad-hoc investigations or to verify that metric
+instrumentation is working correctly. It works with both <xref:System.Diagnostics.Metrics?displayProperty=nameWithType> based
+APIs as well as [EventCounters](event-counters.md).
+
+If the [dotnet-counters](./dotnet-counters.md) tool is not already installed use the SDK to install it:
```

Comment:
>./ [](http://example.com/codeflow?start=25&length=2)

nit: why using `./` here? you didn't use it with the links.

Reply:
I haven't been consistent. My understanding is that both forms work and I think without the './' is more common elsewhere so I am converging on that.

Reply:
In OSS projects I've seen more `./` usage (e.g. https://github.com/open-telemetry/opentelemetry-specification/blob/main/README.md).

I guess this was coming from the Linux shell habit - `./a.out foo bar` would run but `a.out foo bar` would not.